### PR TITLE
Makes ranged mobs have a delay on both attack and targetting, to make it a bit more fair to players.

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request
Title.

Some misc optimizations here and there too and a few QOL things.

TESTED. A GOOD BIT. Ranged attacks work, melee works, targetting works, untargetting works, no runtimes, doesnt feel laggy, and feels somewhat fairer.

Mobs are just as deadly as before-you just have a chance to react.

## Changelog
:cl:
balance: Mobs will now warn you before attacking or targetting you.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/superior_animal/attack.dm [hunk 1]
```diff
@@ -58,9 +58,9 @@
 		else
 			return
 
-/mob/living/carbon/superior_animal/proc/OpenFire(firing_target)
+/mob/living/carbon/superior_animal/proc/OpenFire(var/atom/firing_target)
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/superior_animal/attack.dm:61]:
```suggestion
/mob/living/carbon/superior_animal/proc/OpenFire(atom/firing_target)
```
?

Comment (nikothedude) [reply code/modules/mob/living/carbon/superior_animal/attack.dm:61]: thats just what i do, it works

Comment (Trilbyspaceclone) [reply code/modules/mob/living/carbon/superior_animal/attack.dm:61]: accully why is it an atom?

Comment (r4d6) [reply code/modules/mob/living/carbon/superior_animal/attack.dm:61]: Atoms include turfs, objects and mobs.

Basically it supports more kinds of targets than just mobs.

**Pull request merged**